### PR TITLE
Show image pull progress in a separate Echelon node

### LINKED ISSUE #11: Docker Pull Progress
It will be great to show progress of docker pull in console so it's clear why things are not progressing

### PR DESCRIPTION
**Pull request by fkorotkov**

### CODE REVIEW

**Comment by edigaryev** [comment]
I don't think this solves #11, since the progress is still hidden in `ioutil.Discard`.

**Comment by fkorotkov** [comment]
There will be a timer. When you do `docker pull` you have separate progresses for each layer which is also confusing since it's not a percentage or time estimation. I think a timer will be good enough. What do you think?

**Comment by edigaryev** [comment]
> There will be a timer. When you do `docker pull` you have separate progresses for each layer which is also confusing since it's not a percentage or time estimation. I think a timer will be good enough. What do you think?

How are you going to calculate the total time needed to pull the image anyway?

Note that `ImagePull` returns an `io.Reader` that in turn returns JSON messages about each layer's progress separately:

```
{"status":"Pulling from golangci/golangci-lint","id":"latest"}
{"status":"Pulling fs layer","progressDetail":{},"id":"d6ff36c9ec48"}
{"status":"Pulling fs layer","progressDetail":{},"id":"c958d65b3090"}
{"status":"Pulling fs layer","progressDetail":{},"id":"edaf0a6b092f"}
{"status":"Pulling fs layer","progressDetail":{},"id":"80931cf68816"}
{"status":"Pulling fs layer","progressDetail":{},"id":"813643441356"}
{"status":"Pulling fs layer","progressDetail":{},"id":"2c05b70ce67b"}
{"status":"Pulling fs layer","progressDetail":{},"id":"e82ad91b19e9"}
{"status":"Pulling fs layer","progressDetail":{},"id":"5d2e569cefef"}
{"status":"Waiting","progressDetail":{},"id":"813643441356"}
{"status":"Waiting","progressDetail":{},"id":"2c05b70ce67b"}
{"status":"Waiting","progressDetail":{},"id":"e82ad91b19e9"}
{"status":"Waiting","progressDetail":{},"id":"5d2e569cefef"}
{"status":"Waiting","progressDetail":{},"id":"80931cf68816"}
{"status":"Downloading","progressDetail":{"current":81434,"total":7811570},"progress":"[\u003e                                                  ]  81.43kB/7.812MB","id":"c958d65b3090"}
{"status":"Downloading","progressDetail":{"current":101515,"total":9996337},"progress":"[\u003e                                                  ]  101.5kB/9.996MB","id":"edaf0a6b092f"}
{"status":"Downloading","progressDetail":{"current":736262,"total":7811570},"progress":"[====\u003e                                              ]  736.3kB/7.812MB","id":"c958d65b3090"}
[...]
```

 You can combine them to get a total percentage that can be flushed to the log periodically.

**Comment by fkorotkov** [comment]
Ok, let's not claim fixing #11 then.

File: internal/executor/instance/instance.go [hunk 1]
```diff
@@ -122,12 +122,16 @@ func RunDockerizedAgent(ctx context.Context, config *RunConfig, params *Params)
 	}
 
 	if config.DockerOptions.ShouldPullImage(params.Image) {
-		logger.Debugf("pulling image %s", params.Image)
+		dockerPullLogger := logger.Scoped("docker pull")
+		dockerPullLogger.Infof("pulling image %s", params.Image)
 		progress, err := cli.ImagePull(ctx, params.Image, types.ImagePullOptions{})
 		if err != nil {
+			dockerPullLogger.Errorf("Failed to pull %s: $v", params.Image, err)
```
Comment (edigaryev) [inline internal/executor/instance/instance.go:129]:
```suggestion
			dockerPullLogger.Errorf("failed to pull %s: %v", params.Image, err)
```

Comment (fkorotkov) [reply internal/executor/instance/instance.go]: Hold on, this is a user visible error level log

**Review by edigaryev — APPROVED** [review]

**Pull request merged**